x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportEvaluateDataFrameAction.java

```diff
@@ -48,7 +66,9 @@ protected void doExecute(Task task, EvaluateDataFrameAction.Request request,
             listener::onFailure
         );
 
-        EvaluationExecutor evaluationExecutor = new EvaluationExecutor(threadPool, client, request);
+        // Create a collection of settings to be used by evaluation metrics. Once passed to {@link EvaluationExecutor}, it is immutable.
+        Settings settings = Settings.builder().put(MAX_BUCKET_SETTING.getKey(), maxBuckets.get()).build();
```

Comment:
I think it's unusual to build a new `Settings` object that contains just one setting and pass that around in the production code.  It is passed through three layers of other objects, so in the future this could create confusion for maintainers of the innermost classes about what settings are available.  It makes me wonder if to avoid risk of future mistakes you should introduce a new class, say `EvaluationParameters` that has a member for each required parameter, so it will start off with just `maxBuckets`.  (Alternatively `maxBuckets` could just be passed through the three layers as an `int`, but I guess you were trying to future proof this so that fewer changes are needed when a second dynamic setting needs to be passed.  A bespoke class solves this while avoiding confusion about exactly which settings are included.)

Reply:
> It is passed through three layers of other objects, so in the future this could create confusion for maintainers of the innermost classes about what settings are available

Totally agree. Initially I tried to pass the whole `settings` object but I couldn't find a way to fetch the current version of it from `clusterService`.

> you should introduce a new class, say EvaluationParameters that has a member for each required parameter, so it will start off with just maxBuckets.

Good idea, will do that.

> I guess you were trying to future proof this so that fewer changes are needed when a second dynamic setting needs to be passed. 

Exactly, that was my intention.

Reply:
This is now done.
PTAL